static/js/lib/queries/auth.js

```diff
@@ -1,19 +1,72 @@
 // @flow
-import { FLOW_LOGIN } from "../auth"
+import { nthArg } from "ramda"
+import { FLOW_LOGIN, FLOW_REGISTER } from "../auth"
 
-import type { RequestConfig } from "redux-query"
+import type { AuthResponse, AuthResponseRaw } from "../../flow/authTypes"
+
+const transformAuthResult = (
+  result: AuthResponseRaw
+): { auth: AuthResponse } => ({
+  auth: {
+    partialToken: result.partial_token,
+    flow:         result.flow,
+    state:        result.state,
+    errors:       result.errors,
+    redirectUrl:  result.redirect_url,
+    extraData:    {
+      name: result.extra_data.name
+    }
+  }
+})
+
+const updateAuthResult = {
+  auth: nthArg(1)
```

Comment:
Nit: I think it would help to (a) have a brief explanation that this is just going to replace the previous `auth` state with the one returned in `transform`, or (b) create a helper method called `useUpdatedValue`/`useTransformValue` et al that just returns `nthArg(1)` and could be reused in other query configs. This is nice and clean but I could see it causing some minor confusion